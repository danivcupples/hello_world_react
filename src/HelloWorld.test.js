import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorld from './HelloWorld';

describe(HelloWorld, () => {
  //add tests here
  const name = 'Person';
  const mockRemoveGreeting = jest.fn();
  const component = shallow(
    <HelloWorld name={name} removeGreeting={mockRemoveGreeting}/>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <HelloWorld name='Person'/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the supplied name', () => {
    expect(component.text()).toContain(name);
  });

  it('modifies the greeting when frenchify button is clicked', () => {
    component.find('button.frenchify').simulate('click');
    expect(component.text()).toContain('Bonjour');
  });

  it('modifies the greeting when dogify button is clicked', () => {
    component.find('button.dogify').simulate('click');
    expect(component.text()).toContain('Woof Woof');
  });

  it('calls the passed in removeGreeting function when remove button is clicked', ()=> {
    component.find('button.remove').simulate('click');
    expect(mockRemoveGreeting).toBeCalled();
  });
});
