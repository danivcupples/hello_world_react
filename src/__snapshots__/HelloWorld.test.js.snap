// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class HelloWorld extends _react.Component {
  constructor(props) {
    super(props);
    this.state = { greeting: 'Hello' };
    this.frenchify = this.frenchify.bind(this);
    this.dogify = this.dogify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }

  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }

  dogify() {
    this.setState({ greeting: 'Woof Woof' });
  }

  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }

  render() {
    return (
      _react2.default.createElement('div', { className: 'HelloWorld', __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this },
        this.state.greeting, ', ', this.props.name, '!',
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 29 }, __self: this }),
        _react2.default.createElement('button', { className: 'frenchify', onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 30 }, __self: this }, 'Frenchify!'),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 31 }, __self: this }),
        _react2.default.createElement('button', { className: 'dogify', onClick: this.dogify, __source: { fileName: _jsxFileName, lineNumber: 32 }, __self: this }, 'Dogify!'),
        _react2.default.createElement('br', { __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this }),
        _react2.default.createElement('button', { className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }, 'Remove Me')));


  }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorld"
>
  Hello
  , 
  Person
  !
  <br />
  <button
    className="frenchify"
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <br />
  <button
    className="dogify"
    onClick={[Function]}
  >
    Dogify!
  </button>
  <br />
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove Me
  </button>
</div>
`;
